### DISCUSSION TITLE
SPA routing for GitHub Pages

### DISCUSSION THREAD

**Discussion by nickserv** [community/community#4522] [Other Feature Feedback, Questions, & Ideas]
It's very common for modern frontend web apps to use single page rout via HTML5 pushState. However, this requires replacing a 404 not found page with a catch-all hidden redirect to `index.html`, which GitHub Pages does not support. This is an unfortunate limitation as it means I either have to use older routers such as hash routers (which aren't even fully supported by React Router any more) or find an alternative CDN such as Netlify just for this feature.

**Comment by PeterHewat** [comment]
I wouldn't hold my breath on this one (however simple it may be to implement) since this request has been lingering for over 6 years now...
https://github.com/isaacs/github/issues/408
